core/src/com/unciv/models/ruleset/validation/RulesetErrorList.kt

```diff
@@ -60,4 +91,21 @@ class RulesetErrorList : ArrayList<RulesetError>() {
                         // out of place. Prevent via kludge:
                         it.text.replace('<','〈').replace('>','〉')
                 }
+
+    companion object {
+        private val empty = RulesetErrorList()
+
+        fun of(ruleset: Ruleset? = null, sourceObject: IHasUniques? = null, vararg errors: RulesetError): RulesetErrorList {
+            if (errors.isEmpty()) return empty
+            val result = RulesetErrorList(ruleset)
+            for (error in errors)
+                result.add(sourceObject, error)
+            return result
+        }
+        fun of(ruleset: Ruleset? = null, sourceObject: IHasUniques? = null, text: String, severity: RulesetErrorSeverity = RulesetErrorSeverity.Error): RulesetErrorList {
```

Comment:
Putting the only required parameter as the third parameter instead of the first feels like an antipattern